02-feature-engineering.Rmd

```diff
@@ -6,9 +6,9 @@ source("_common.R")
+Feature engineering is a fancy machine learning way of saying "prepare your data for analysis". But it's also more than just getting your data in the right format. It's about transforming variables to help the model provide more stable predictions -- for example, encoding, modeling, or omitting missing data; creating new variables from the available dataset; and otherwise helping improve the model performance through transformations and modifications to the existing data *before* analysis.
 
-One of the biggest challenges with feature engineering is what's referred to as **data leakage**, which is when information from the test dataset leaks into the training dataset. Consider the simple example of normalizing or standardizing a variable (subtracting the variable mean from each value and then dividing by the standard deviation). Normalizing your variables is neccesary for a wide range of predictive models, including any models using regularization methods, principal components analysis, and $k$-nearest neighbor models, among others. But when we normalize the variable, it is critical we do so relative to the training set, *not* relative to the full dataset. If we normalize the numeric variables in our dataset, then divide it into test and training datasets, then information from the test dataset has *leaked* into the training dataset. This seems simple enough - just wait to standardize until after splitting - but it becomes more complicated when we consider *tuning* a model. If we use a process like $k$-fold cross-validation, then we have to normalize *within* each fold to get an accurate representation of out-of-sample predictive accuracy.
+One of the biggest challenges with feature engineering is what's referred to as **data leakage**, which occurs when information from the test dataset leaks into the training dataset. Consider the simple example of normalizing or standardizing a variable (subtracting the variable mean from each value and then dividing by the standard deviation). Normalizing your variables is necessary for a wide range of predictive models, including any model using regularization methods, principal components analysis, and $k$-nearest neighbor models, among others. But when we normalize the variable, it is critical we do so relative to the training set, *not* relative to the full dataset. If we normalize the numeric variables in our full dataset and then divide it into test and training datasets, information from the test dataset has *leaked* into the training dataset. This seems simple enough - just wait to standardize until after splitting - but it becomes more complicated when we consider *tuning* a model. If we use a process like $k$-fold cross-validation, then we have to normalize *within* each fold to get an accurate representation of out-of-sample predictive accuracy.
 
 <!-- In the above, things like k-fold cv will probably have chapters of their own. We should link to them in the text -->
 
```

Comment:
This is a minor technical note, but it looks like to be consistent with the tidyverse style guide we should refer to package names without curly braces, e.g. `recipes` instead of `{recipes}`.  See[ section 7.7](https://style.tidyverse.org/documentation.html#r-code), specifically, 

> "Do not use code font for package names. If the package name is ambiguous in the context, disambiguate with words, e.g. “the foo package”.

I personally prefer having the curly braces because it makes it clear that you're referring to a package name, but I'll defer to your opinion here!

Reply:
I've thought about this too. It looks like that quote is specifically about using the code font though, right? So `code` as opposed to code. I like the curly braces and my favorite thing to do is to link to the package documentation, but that gets a bit burdensome to do every time you refer to a package. 

What if we say we'll stick with the curly braces and have it be italicized, with the first mention in a chapter linking to the documentation. So, the first time it would be [*{recipes}*](https://recipes.tidymodels.org) and then from there on out *{recipes}* for each chapter (but we link to the documentation the first time in each chapter).

Reply:
Sounds good! I've been perusing some other books (e.g. https://www.tmwr.org/recipes.html), and it looks like others bold package names, but I don't think that's a universal convention, so let's go with what you proposed! I'll make those changes as I go through the rest of the chapter.

Reply:
I'm happy to go with bolding instead of italics. Either works for me!

Reply:
Okie doke, I'll go with bolding and curly braces. Sorry for the nit-picking! 🙃

Reply:
All good!